pkg/graphql/request.go

```diff
@@ -110,8 +109,14 @@ func (r *Request) IsNormalized() bool {
 	return r.isNormalized
 }
 
-func (r *Request) IsValidated() bool {
-	return r.isValidated
+func (r *Request) IsValidated(schema *Schema) bool {
+	schemaHash, _ := schema.Hash()
+	return r.isValid(schemaHash)
+}
+
+func (r *Request) isValid(schemaHash uint64) bool {
+	result := r.validForSchema[schemaHash]
```

Comment:
I think we do have a nil pointer danger here, it may be better to check if the hash exists, like this:

```
result, ok := r.validForSchema[schemaHash]
if !ok {
  // do something when the hash does not exist
}
```